core/services/feeds/service.go

```diff
@@ -254,8 +258,45 @@ func (s *service) RegisterManager(ctx context.Context, params RegisterManagerPar
 	return id, nil
 }
 
-// SyncNodeInfo syncs the node's information with FMS
+// syncNodeInfoWithRetry syncs the node's information with FMS using a goroutine.
+// In case of failures, it retries with an exponential backoff for up to 24h.
+func (s *service) syncNodeInfoWithRetry(id int64) {
+	// cancel the previous context -- and, by extension, the existing goroutine --
+	// so that we can start anew
+	s.syncNodeInfoCancel()
```

Comment:
I dont think we need to do this right?

If the caller of `syncNodeInfoWithRetry` pass in their context which is scoped to a request, then we dont have to manually cancel each context. Each request should have its own retry. Eg request A should not cancel request B sync which is happening with this setup?

Reply:
> Each request should have its own retry. Eg request A should not cancel request B sync which is happening with this setup?


Well, that's actually by design. The idea behind the implementation is that there should only be one retry goroutine running. The main reason I chose to go that path is to avoid overloading the JD server in case of several requests that might get queued for a retry.

Even though the retry intervals are relatively large now, I don't think it's hard to imagine a scenario where a user repeats an action a few dozen times while the JD service is offline, which could translate into dozens or even hundreds of simultaneous requests to JD. And if you've followed the profiling and optimization work we did a few weeks ago on CLO, you should know that his would not end well.

Performance reasons aside, I don't think we can use the request context for the retries. I did it before but what I noticed in those tests is that the request context is canceled -- and the associated retries aborted -- as soon as we return a response to the http client.

Reply:
> I don't think we can use the request context for the retries. I did it before but what I noticed in those tests is that the request context is canceled -- and the associated retries aborted -- as soon as we return a response to the http client.

ah yes good point, we cant use the request context then!

Reply:
> The idea behind the implementation is that there should only be one retry goroutine running.
I think this is not entirely correct?

So is it `one retry goroutine per feeds manager connection` or `one retry goroutine per whole system` , what i am seeing in this PR is the latter, i thought we want the first one ?  If that is the case, then we dont need to worry about the deadlock issue since we wont be sharing context between different feed manager connections.

Reply:
You're correct, should be `one retry goroutine per feeds manager connection`. I still haven't assimilated the idea that there may be multiple feed managers. 😞 

I'll need to revisit the solution with that in mind.